studio/pages/sign-in.tsx

```diff
@@ -8,13 +8,28 @@ import { SignInLayout } from 'components/layouts'
 import { useEffect } from 'react'
 import { IS_PLATFORM } from 'lib/constants'
 import { useRouter } from 'next/router'
+import { auth } from 'lib/gotrue'
 
 const SignInPage: NextPageWithLayout = () => {
   const router = useRouter()
   useEffect(() => {
     // if the dashboard is running locally, redirect straight to the projects page
     if (!IS_PLATFORM) {
       router.replace('/projects')
+      return
+    }
+
+    const {
+      data: { subscription },
+    } = auth.onAuthStateChange((event) => {
+      if (event === 'SIGNED_IN') {
```

Comment:
@hf to confirm, the `SIGNED_IN` event only fires from other tabs? Not from the same tab after a sign in?

Reply:
i just checked this locally - the signed in event doesnt fire on the same tab for the dash cause the browser refreshes after logging in

Reply:
This uses [BroadcastChannel](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API) in the background, so cross-tab events will be listened to. 

But, I see your point about this firing once a user clicks the sign in button. Not sure if that causes any issues, as I was not able to spot them really.

Reply:
i think this is okay really - since this is how pub/sub with the broadcast channel works, cant think of any other better way as well